src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -316,6 +316,42 @@ internal static unsafe void FixupMethodCell(IntPtr hModule, MethodFixupCell* pCe
             }
         }
 
+#if !PLATFORM_UNIX
+        private static unsafe IntPtr GetProcAddress(IntPtr hModule, byte* methodName, CharSet charSetMangling)
+        {
+            // First look for the unmangled name.  If it is unicode function, we are going
+            // to need to check for the 'W' API because it takes precedence over the
+            // unmangled one (on NT some APIs have unmangled ANSI exports).
+            
+            var exactMatch = Interop.mincore.GetProcAddress(hModule, methodName);
+
+            if ((charSetMangling == CharSet.Ansi && exactMatch != IntPtr.Zero) || charSetMangling == 0)
+            {
+                return exactMatch;
+            }
+
+            int nameLength = strlen(methodName);
+
+            // We need to add an extra byte for the suffix, and an extra byte for the null terminator
+            byte* probedMethodName = stackalloc byte[nameLength + 2];
+
+            for (int i = 0; i < nameLength; i++)
+            {
+                probedMethodName[i] = methodName[i];
+            }
+
+            probedMethodName[nameLength] = (charSetMangling == CharSet.Ansi) ? (byte)'A' : (byte)'W';
```

Comment:
@jkotas , I think this relies on stackalloc doing zero-initialization for a null terminator. Is that a safe assumption in CoreLib?

Reply:
The zero initialization was not turned off in CoreRT CoreLib yet, but I think we will want to do that in future.

It would be nice to write the code with assumption that stackalloc memory is not zero initialized.